src/plugin/fee-tracking.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * fee-tracking.ts
+ *
+ * API key fee tracking for rate limiting.
+ */
+
+import { PluginKVStore, pluginError } from '@openzeppelin/relayer-sdk';
+import { HTTP_STATUS } from './constants';
+
+export class FeeTracker {
+  constructor(
+    private readonly kv: PluginKVStore,
+    private readonly network: 'testnet' | 'mainnet',
+    private readonly apiKey: string,
+    private readonly feeLimit: number
+  ) {}
+
+  private get key(): string {
+    return FeeTracker.buildKey(this.network, this.apiKey);
+  }
+
+  static buildKey(network: 'testnet' | 'mainnet', apiKey: string): string {
+    return `${network}:api-key-fees:${apiKey}`;
+  }
+
+  /**
+   * Get fee consumption for an API key (for management API).
+   */
+  static async getConsumed(
+    kv: PluginKVStore,
+    network: 'testnet' | 'mainnet',
+    apiKey: string
+  ): Promise<number> {
+    const key = FeeTracker.buildKey(network, apiKey);
+    const data = await kv.get<{ consumed: number }>(key);
+    return data?.consumed ?? 0;
+  }
+
+  /**
+   * Check if the API key has exceeded its fee limit.
+   * Throws 429 if limit exceeded.
+   */
+  async checkLimit(): Promise<void> {
```

Comment:
Do we check against fee for new tx as well?

Let's say that user has 1 stroop remaining. He would pass this check.

Do we check at some point consumed + tx fee >= limit